src/Compilers/Test/Utilities/CSharp/CSharpTestSource.cs

```diff
@@ -27,12 +30,34 @@ private CSharpTestSource(object value)
             Value = value;
         }
 
+        public static SyntaxTree Parse(
+            string text,
+            string path = "",
+            CSharpParseOptions options = null,
+            Encoding encoding = null,
+            SourceHashAlgorithm checksumAlgorithm = SourceHashAlgorithms.Default)
+        {
+            var stringText = SourceText.From(text, encoding ?? Encoding.UTF8, checksumAlgorithm);
+            var tree = SyntaxFactory.ParseSyntaxTree(stringText, options ?? TestOptions.RegularPreview, path);
+            CheckSerializable(tree);
+            return tree;
+        }
+
+        private static void CheckSerializable(SyntaxTree tree)
+        {
+            using var stream = new MemoryStream();
+            var root = tree.GetRoot();
+            root.SerializeTo(stream);
+            stream.Position = 0;
+            CSharpSyntaxNode.DeserializeFrom(stream);
```

Comment:
Should this be doing some sort of comparison? If not, is the goal just that the deserialization doesn't crash? If so, comment might help here?

Reply:
This code just moved from elsewhere.